lib/ex_aws/auth.ex

```diff
@@ -114,9 +123,13 @@ defmodule ExAws.Auth do
   end
 
   defp canonical_query_params(nil), do: ""
-  defp canonical_query_params(params) do
+  defp canonical_query_params(params) when is_binary(params) do
+    params
+    |> String.split("&") |> Enum.map(&List.to_tuple(String.split(&1, "=")))
+    |> canonical_query_params
+  end
```

Comment:
When might it be binary? I'm not really a fan of having a basic URI params parser here. Doesn't URI.parse do this anwyay?